repo/repo.go

```diff
@@ -23,6 +23,9 @@ type Repo interface {
 	// to the returned config are not automatically persisted.
 	Config() (*config.Config, error)
 
+	// Path is the repo file-system path
+	Path() string
```

Comment:
What about in-memory or remote / virtual repos?

Reply:
Currently only FSRepo is provided. For the other kind of repo implementations, there should be design to be compatible with resourcemanager or nopfs.
https://github.com/ipfs/kubo/blob/ce23fc7cccf1bea7ba0e93900b3654d2b879cad9/repo/fsrepo/fsrepo.go#L366